samza-kafka/src/main/scala/org/apache/samza/checkpoint/kafka/KafkaCheckpointManager.scala

```diff
@@ -88,21 +88,26 @@ class KafkaCheckpointManager(
     val msgBytes = serde.toBytes(checkpoint)
     retryBackoff.run(
       loop => {
-        if (producer == null) {
-          producer = connectProducer()
+        if (systemProducer == null) {
+          systemProducer = getSystemProducer()
+          systemProducer.register(taskName.getTaskName)
+          systemProducer.start
         }
 
-        producer.send(new ProducerRecord(checkpointTopic, 0, keyBytes, msgBytes)).get()
+        val systemStream = new SystemStream(systemName, checkpointTopic)
+        val envelope: OutgoingMessageEnvelope = new OutgoingMessageEnvelope(systemStream, keyBytes, msgBytes)
+        systemProducer.send(taskName.getTaskName, envelope)
+        systemProducer.flush(taskName.getTaskName) // make sure it is written
         loop.done
       },
 
       (exception, loop) => {
         warn("Failed to write %s partition entry %s: %s. Retrying." format(CHECKPOINT_LOG4J_ENTRY, key, exception))
         debug("Exception detail:", exception)
-        if (producer != null) {
-          producer.close
+        if (systemProducer != null) {
+          systemProducer.stop
         }
-        producer = null
+        systemProducer = null
```

Comment:
Given that we are no longer using the `SimpleConsumer` can we get rid off the `getConsumer` code completely? that will simplify this class further

Reply:
done.